src/binder/binder.cpp

```diff
@@ -26,7 +27,11 @@ std::unique_ptr<BoundStatement> Binder::bind(const Statement& statement) {
     std::unique_ptr<BoundStatement> boundStatement;
     switch (statement.getStatementType()) {
     case StatementType::CREATE_TABLE: {
-        boundStatement = bindCreateTable(statement);
+        if (statement.constPtrCast<CreateTable>()->getSource()) {
+            boundStatement = bindCreateTableAs(statement);
+        } else {
+            boundStatement = bindCreateTable(statement);
```

Comment:
I would move this check in bindCreateTable so inside it check if you have a source and need to trigger bindCreateTableAs

Reply:
Done with https://github.com/kuzudb/kuzu/pull/5354/commits/59208ac94ae481ece5cd523e2005e40246f0fb0a